tests/h/models/user_identity_test.py

```diff
@@ -76,18 +93,76 @@ def test_two_cant_have_the_same_provider_and_provider_id(self, db_session):
         ):
             db_session.flush()
 
-    def test_two_can_have_the_same_provider_id_but_different_providers(
-        self, db_session
+    def test_one_user_can_have_the_same_provider_id_from_different_providers(
+        self, db_session, user
     ):
         db_session.add_all(
             [
-                models.UserIdentity(provider="provider_1", provider_unique_id="id"),
-                models.UserIdentity(provider="provider_2", provider_unique_id="id"),
+                models.UserIdentity(
+                    provider="provider_1", provider_unique_id="id", user=user
+                ),
+                models.UserIdentity(
+                    provider="provider_2", provider_unique_id="id", user=user
+                ),
             ]
         )
 
         db_session.flush()
 
+    def test_different_users_can_have_the_same_provider_id_from_different_providers(
+        self, db_session, factories
+    ):
+        db_session.add_all(
+            [
+                models.UserIdentity(
+                    provider="provider_1",
+                    provider_unique_id="id",
+                    user=factories.User(),
+                ),
+                models.UserIdentity(
+                    provider="provider_2",
+                    provider_unique_id="id",
+                    user=factories.User(),
+                ),
+            ]
+        )
+
+        db_session.flush()
+
+    def test_removing_a_user_identity_from_a_user_deletes_it(self, db_session, user):
+        # Add a couple of noise UserIdentity's. These should not be removed
+        # from the DB.
+        models.UserIdentity(provider="provider", provider_unique_id="1", user=user)
+        models.UserIdentity(provider="provider", provider_unique_id="2", user=user)
+        # The UserIdentity that we are going to remove.
+        user_identity = models.UserIdentity(
+            provider="provider", provider_unique_id="3", user=user
+        )
+
+        user.identities.remove(user_identity)
+
+        assert user_identity not in db_session.query(models.UserIdentity).all()
+
+    def test_deleting_a_user_identity_removes_it_from_its_user(self, db_session, user):
```

Comment:
I don't understand how this test inspects the `user` ... I'm only seeing references to `user_identity` here in the assertion.

Reply:
Hmm, that's a good point. The problem is that [in the tests the `UserIdentity` _won't_ be removed from `user.identities`](https://github.com/hypothesis/h/pull/5097#issuecomment-399940134) because of what seems to be an issue with our `db_session` test fixture. So I changed the test to use `was_deleted` instead. But now it's totally not testing what it says it tests anymore

Reply:
Ok, fixed this test